src/el-data-table.vue

```diff
@@ -375,9 +367,7 @@ export default {
      */
     beforeSearch: {
       type: Function,
-      default() {
-        return Promise.resolve()
-      }
+      default() {}
```

Comment:
```suggestion
      async default() {}
```
这样是不是会好一点？不知道这样写 ok 不

Reply:
这样可以比较直观的知道这是异步的

Reply:
重构之后，不再要求 beforeSearch 返回一个 Promise。现在 beforeSearch 是同步的也可以